core/controllers/library_test.py

```diff
@@ -48,6 +52,63 @@ def test_library_page(self):
         response = self.get_html_response(feconf.LIBRARY_INDEX_URL)
         response.mustcontain('ng-controller="Library"')
 
+    def test_library_handler_with_exceeding_query_limit_logs_error(self):
+        response_dict = self.get_json(feconf.LIBRARY_SEARCH_DATA_URL)
+        self.assertEqual({
+            'iframed': False,
+            'is_admin': False,
+            'is_topic_manager': False,
+            'is_moderator': False,
+            'is_super_admin': False,
+            'activity_list': [],
+            'additional_angular_modules': [],
+            'search_cursor': None
+        }, response_dict)
+
+        # Load a public demo exploration.
+        exp_services.load_demo('0')
+
+        observed_log_messages = []
+
+        def _mock_logging_function(msg, *_):
+            """Mocks logging.error()."""
+            observed_log_messages.append(msg)
+
+        logging_swap = self.swap(logging, 'error', _mock_logging_function)
+        default_query_limit_swap = self.swap(feconf, 'DEFAULT_QUERY_LIMIT', 1)
+        # Load the search results with an empty query.
+        with default_query_limit_swap, logging_swap:
+            self.get_json(feconf.LIBRARY_SEARCH_DATA_URL)
+
+            self.assertEqual(len(observed_log_messages), 1)
+            self.assertEqual(
+                observed_log_messages[0],
+                (
+                    '1 activities were fetched to load the library page. '
```

Comment:
No need to use parenthesis here, just use Python's automatic string-literal concatenation.